src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,485 @@
+                    }
+                } while (!validInput);
+
+                try {
+                    storeFile.saveFile(tasks);
+                } catch (DukeException e) {
+                    ui.printMsg(ui.showErrorMsg(e));
+                }
+
+                break;
+            case "delete":
+                if (tasks.isEmpty()) {
+                    ui.printMsg(ui.showListEmptyMsg());
+                } else {
+                    ui.printMsg(ui.showListMsg());
+                    ui.printMsg(ui.showTaskList(tasks));
+                    do {
+                        ui.printMsg(ui.showTaskDeleteAskMsg());
+                        try {
+                            userInput = ui.readInput(isGui);
+                            TaskList deletedTasks = new TaskList();
+                            int[] tasksArray = Parser.parse(userInput);
+                            for (int index : tasksArray) {
+                                try {
+                                    if (index > tasks.size() || index <= 0) {
+                                        throw new DukeException("This task #" + index + " does not exist.");
+                                    }
+                                    Task task = tasks.getTask(index);
+                                    tasks.removeTask(index);
+                                    deletedTasks.addTask(task);
+                                } catch (DukeException e) {
+                                    ui.printMsg(ui.showErrorMsg(e));
+                                }
+                            }
+                            if (!deletedTasks.isEmpty()) {
+                                ui.printMsg(ui.showTaskDeleteMsg());
+                                ui.printMsg(ui.showTaskList(deletedTasks));
+                            }
+                            validInput = true;
+                        } catch (DukeException e) {
+                            ui.printMsg(ui.showErrorMsg(e));
+                            validInput = false;
+                        }
+                    } while (!validInput);
+                }
+
+                try {
+                    storeFile.saveFile(tasks);
+                } catch (DukeException e) {
+                    ui.printMsg(ui.showErrorMsg(e));
+                }
+
+                break;
+            case "help":
+                ui.printMsg(ui.showHelp());
+                break;
+            default:
+                try {
+                    throw new DukeException("Unrecognised Command :(, type 'help' for available commands.");
+                } catch (DukeException e) {
+                    ui.printMsg(ui.showErrorMsg(e));
+                }
+            }
+        }
+        ui.printMsg(ui.showByeMsg());
+    }
+
+    /**
+     * Main execution of Duke Chat bot for GUI
+     */
+    public void runGui() {
+        String[] GuiInputArray;
+        String[] userInputArray;
+
+        userInput = ui.readInput(isGui);
```

Comment:
Again, i think you can abstract all the methods out so that your code is easier to read!